azure-iot-device/tests/common/pipeline/test_pipeline_stages_mqtt.py

```diff
@@ -388,33 +361,39 @@ def test_provider_function_throws_exception_2(
         assert handler_before == handler_after
 
 
-@pytest.mark.describe("MQTT Provider pipeline stage events from transport library")
+@pytest.mark.describe("Provider - .on_mqtt_message-received()")
```

Comment:
So, again, I'm confused by what this is.

It SEEMS to me, after looking into the test, that you're testing what happens when the `MQTTProvider` triggers it's `.on_mqtt_message_received` handler, thus triggering the `Provider` method `_on_message_received`.

- This `.describe` refers to the wrong method name
- This should probably use the `EVENT:` syntax, instead of the method syntax, right? After all, we're testing what happens in response to an external event using handlers, and the handler isn't the unit under test because it's convention-private. The unit under test is the external event, and what happens in response (the convention private `_on_message_received` is implicitly tested by testing the event)
- I keep repeating this, but `Provider` really should be made into `ProviderStage` to reduce confusion
-